lib/app/routes/exercises.rb

```diff
@@ -133,6 +141,51 @@ class Exercises < Core
         Hack::UpdatesUserExercise.new(submission.user_id, submission.track_id, submission.slug).update
         redirect "/"
       end
+
+      get '/submissions/blob/content' do
+        content_type :json
+        submission = Submission.find_by_key(params[:key])
+        blob = Octokit.blob("#{submission.user.username}/#{submission.slug}", params[:sha])
+        s = SourceClassifier.new(File.join(File.dirname(__FILE__), '../../../bin/', 'trainer.bin'))
+        result =  Base64.decode64(blob.content)
+        source_language = s.identify(result)
+        marked_content = ConvertsMarkdownToHTML.convert("```#{source_language.downcase}\n#{result}\n```")
+        content = { data: marked_content } 
+        content.to_json
+      end
+
+      def get_tree(submission)
+        Octokit.configure do |c|
+          c.login = 'SaiPramati'
+          c.password = 'pramati123'
+        end
+        git_tree_source = Octokit.tree("#{submission.user.username}/#{submission.slug}",
+                                        submission.solution.values.first, recursive: true)
```

Comment:
Line is too long. [90/80]<br>Align the parameters of a method call if they span more than one line.